Index: src/main/java/uk/gov/hmcts/probate/controller/DocumentsController.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/uk/gov/hmcts/probate/controller/DocumentsController.java	(revision 1a78c098d76c9b6dcb1339ca978ed2f3de449dd8)
+++ src/main/java/uk/gov/hmcts/probate/controller/DocumentsController.java	(date 1548417573000)
@@ -5,7 +5,11 @@
 import org.springframework.http.HttpStatus;
 import org.springframework.http.MediaType;
 import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.*;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestController;
 import uk.gov.hmcts.probate.service.BusinessService;
 import uk.gov.hmcts.reform.probate.model.documents.CheckAnswersSummary;
 
Index: src/main/java/uk/gov/hmcts/probate/client/BusinessServiceConfiguration.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/uk/gov/hmcts/probate/client/BusinessServiceConfiguration.java	(revision 1a78c098d76c9b6dcb1339ca978ed2f3de449dd8)
+++ src/main/java/uk/gov/hmcts/probate/client/BusinessServiceConfiguration.java	(date 1548672610000)
@@ -1,7 +1,6 @@
 package uk.gov.hmcts.probate.client;
 
 import com.fasterxml.jackson.databind.ObjectMapper;
-import com.fasterxml.jackson.databind.SerializationFeature;
 import feign.codec.Decoder;
 import feign.jackson.JacksonDecoder;
 import org.springframework.context.annotation.Bean;
@@ -14,7 +13,6 @@
     @Bean
     @Primary
     Decoder feignDecoder(ObjectMapper objectMapper) {
-        objectMapper.enable(SerializationFeature.WRAP_ROOT_VALUE);
         return new JacksonDecoder(objectMapper);
     }
 }
Index: src/main/java/uk/gov/hmcts/probate/client/SubmitServiceApi.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/uk/gov/hmcts/probate/client/SubmitServiceApi.java	(revision 1a78c098d76c9b6dcb1339ca978ed2f3de449dd8)
+++ src/main/java/uk/gov/hmcts/probate/client/SubmitServiceApi.java	(date 1548763506000)
@@ -9,6 +9,7 @@
 import org.springframework.web.bind.annotation.RequestParam;
 import uk.gov.hmcts.reform.probate.model.cases.ProbateCaseDetails;
 import uk.gov.hmcts.reform.probate.model.cases.ProbatePaymentDetails;
+import uk.gov.hmcts.reform.probate.model.cases.SubmitResult;
 
 import static org.springframework.http.HttpHeaders.AUTHORIZATION;
 import static org.springframework.http.HttpHeaders.CONTENT_TYPE;
@@ -47,7 +48,7 @@
             value = "/submissions/{applicantEmail}",
             headers = CONTENT_TYPE + "=" + APPLICATION_JSON_VALUE
     )
-    ProbateCaseDetails submit(
+    SubmitResult submit(
             @RequestHeader(AUTHORIZATION) String authorisation,
             @RequestHeader(SubmitServiceConfiguration.SERVICE_AUTHORIZATION) String serviceAuthorization,
             @PathVariable(SubmitServiceConfiguration.APPLICANT_EMAIL) String applicantEmail,
Index: src/main/java/uk/gov/hmcts/probate/core/service/SubmitServiceImpl.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/uk/gov/hmcts/probate/core/service/SubmitServiceImpl.java	(revision 1a78c098d76c9b6dcb1339ca978ed2f3de449dd8)
+++ src/main/java/uk/gov/hmcts/probate/core/service/SubmitServiceImpl.java	(date 1548763506000)
@@ -15,6 +15,7 @@
 import uk.gov.hmcts.reform.probate.model.cases.CasePayment;
 import uk.gov.hmcts.reform.probate.model.cases.ProbateCaseDetails;
 import uk.gov.hmcts.reform.probate.model.cases.ProbatePaymentDetails;
+import uk.gov.hmcts.reform.probate.model.cases.SubmitResult;
 import uk.gov.hmcts.reform.probate.model.forms.CcdCase;
 import uk.gov.hmcts.reform.probate.model.forms.Form;
 import uk.gov.hmcts.reform.probate.model.forms.Payment;
@@ -70,13 +71,13 @@
         assertEmail(applicantEmail, form);
         FormMapper formMapper = mappers.get(form.getType());
         log.debug("calling submit on submitserviceapi");
-        ProbateCaseDetails probateCaseDetails = submitServiceApi.submit(
+        SubmitResult submitResult = submitServiceApi.submit(
             securityUtils.getAuthorisation(),
             securityUtils.getServiceAuthorisation(),
             applicantEmail,
             ProbateCaseDetails.builder().caseData(mapToCase(form, formMapper)).build()
         );
-        return mapFromCase(formMapper, probateCaseDetails);
+        return mapFromCase(formMapper, submitResult.getProbateCaseDetails());
     }
 
     private void assertEmail(String applicantEmail, Form form) {
Index: src/test/java/uk/gov/hmcts/probate/core/service/mapper/IntestacyTestDataCreator.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/test/java/uk/gov/hmcts/probate/core/service/mapper/IntestacyTestDataCreator.java	(revision 1a78c098d76c9b6dcb1339ca978ed2f3de449dd8)
+++ src/test/java/uk/gov/hmcts/probate/core/service/mapper/IntestacyTestDataCreator.java	(date 1548413031000)
@@ -159,7 +159,7 @@
     public static GrantOfRepresentationData createGrantOfRepresentation() {
         GrantOfRepresentationData grantOfRepresentation = new GrantOfRepresentationData();
         grantOfRepresentation.setApplicationType(ApplicationType.PERSONAL);
-        grantOfRepresentation.setCaseType(GrantType.INTESTACY);
+        grantOfRepresentation.setGrantType(GrantType.INTESTACY);
         grantOfRepresentation.setPrimaryApplicantEmailAddress(EMAIL);
         grantOfRepresentation.setPrimaryApplicantForenames(FIRST_NAME);
         grantOfRepresentation.setPrimaryApplicantSurname(LAST_NAME);
Index: src/test/java/uk/gov/hmcts/probate/core/service/mapper/IntestacyMapperTest.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/test/java/uk/gov/hmcts/probate/core/service/mapper/IntestacyMapperTest.java	(revision 1a78c098d76c9b6dcb1339ca978ed2f3de449dd8)
+++ src/test/java/uk/gov/hmcts/probate/core/service/mapper/IntestacyMapperTest.java	(date 1548413031000)
@@ -70,7 +70,7 @@
     public void shouldMapEmptyIntestacyFormToGrantOfRepresentation() {
         GrantOfRepresentationData expectedGrantOfRepresentation = new GrantOfRepresentationData();
         expectedGrantOfRepresentation.setApplicationType(ApplicationType.PERSONAL);
-        expectedGrantOfRepresentation.setCaseType(GrantType.INTESTACY);
+        expectedGrantOfRepresentation.setGrantType(GrantType.INTESTACY);
         GrantOfRepresentationData actualGrantOfRepresentation = mapper.toCaseData(new IntestacyForm());
         Assert.assertThat(actualGrantOfRepresentation, equalTo(expectedGrantOfRepresentation));
         assertThat(actualGrantOfRepresentation).isEqualToComparingFieldByFieldRecursively(expectedGrantOfRepresentation);
Index: src/contractTest/resources/json/intestacyGrantOfRepresentation_partial_draft.json
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/contractTest/resources/json/intestacyGrantOfRepresentation_partial_draft.json	(revision 1a78c098d76c9b6dcb1339ca978ed2f3de449dd8)
+++ src/contractTest/resources/json/intestacyGrantOfRepresentation_partial_draft.json	(date 1548757631000)
@@ -1,6 +1,7 @@
 {
   "caseData": {
     "type": "GrantOfRepresentation",
+    "caseType":"intestacy",
     "applicationType": "Personal",
     "primaryApplicantEmailAddress": "jsnow@bbc.co.uk",
     "primaryApplicantForenames": "Jon",
@@ -17,7 +18,7 @@
     "deceasedDateOfDeath": "2018-01-01"
   },
   "caseInfo": {
-    "caseId": "caseId",
-    "state": "state"
+    "caseId": "12323213323",
+    "state": "Draft"
   }
 }
Index: src/contractTest/resources/json/intestacyGrantOfRepresentation_invalid_response.json
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/contractTest/resources/json/intestacyGrantOfRepresentation_invalid_response.json	(revision 1a78c098d76c9b6dcb1339ca978ed2f3de449dd8)
+++ src/contractTest/resources/json/intestacyGrantOfRepresentation_invalid_response.json	(date 1548764808000)
@@ -1,94 +1,70 @@
 {
-  "path":"/submissions/email",
-  "timestamp":"2019-01-03T15:31:16.525+0000",
-  "errors":[
-    {
-      "field":"caseData",
-      "bindingFailure":false,
-      "code":"AssertExpression",
-      "codes":[
-        "AssertExpression.probateCaseDetails.caseData",
-        "AssertExpression.caseData",
-        "AssertExpression.uk.gov.hmcts.reform.probate.model.cases.CaseData",
-        "AssertExpression"
-      ],
-      "defaultMessage":"deceasedDateOfBirth.isBefore(deceasedDateOfDeath) is false",
-      "rejectedValue":{
-        "primaryApplicantPhoneNumber":"123455678",
-        "deceasedFreeTextAddress":"Winterfell, Westeros",
-        "deceasedAddressFound":"Yes",
-        "primaryApplicantAdoptionInEnglandOrWales":"Yes",
-        "uploadDocumentUrl":"http://document-management/document/12345",
-        "deceasedSpouseNotApplyingReason":"mentallyIncapable",
-        "declaration":{
-          "declarationCheckbox":"Yes"
-        },
-        "primaryApplicantFreeTextAddress":"Pret a Manger St. Georges Hospital Blackshaw Road",
-        "deceasedAnyDeceasedChildrenDieBeforeDeceased":"No",
-        "ihtFormCompletedOnline":"Yes",
-        "deceasedAliasNameList":[
-          {
-            "value":{
-              "Forenames":"King",
-              "LastName":"North"
-            }
-          }
-        ],
-        "ihtGrossValue":"100000",
-        "applicationType":"Intestacy",
-        "ihtNetValue":"100000",
-        "deceasedAnyOtherNames":"No",
-        "ihtReferenceNumber":"GOT123456",
-        "deceasedHasAssetsOutsideUk":"Yes",
-        "deceasedAnyChildren":"No",
-        "outsideUKGrantCopies":6,
-        "primaryApplicantRelationshipToDeceased":"adoptedChild",
-        "extraCopiesOfGrant":5,
-        "deceasedForenames":"Ned",
-        "primaryApplicantEmailAddress":"jsnow@bbc.co.uk",
-        "primaryApplicantSurname":"Snow",
-        "primaryApplicantForenames":"Jon",
-        "primaryApplicantAddress":{
-          "AddressLine1":"Pret a Manger St. Georges Hospital Blackshaw Road London SW17 0QT"
-        },
-        "deceasedAddress":{
-          "AddressLine1":"Winterfell, Westeros"
-        },
-        "deceasedDivorcedInEnglandOrWales":"No",
-        "deceasedMartialStatus":"marriedCivilPartnership",
-        "deceasedDateOfBirth":"2018-01-01",
-        "assetsOverseasNetValue":"10050",
-        "primaryApplicantAddressFound":"Yes",
-        "deceasedOtherChildren":"Yes",
-        "ihtFormId":"IHT205",
-        "registryLocation":"Birmingham",
-        "childrenOverEighteenSurvived":"Yes",
-        "deceasedSurname":"Stark",
-        "grandChildrenSurvivedUnderEighteen":"No",
-        "deceasedDateOfDeath":"1920-01-01"
-      },
-      "arguments":[
-        {
-          "codes":[
-            "probateCaseDetails.caseData",
-            "caseData"
-          ],
-          "arguments":null,
-          "defaultMessage":"caseData",
-          "code":"caseData"
-        },
-        {
-          "arguments":null,
-          "codes":[
-            "deceasedDateOfBirth.isBefore(deceasedDateOfDeath)"
-          ],
-          "defaultMessage":"deceasedDateOfBirth.isBefore(deceasedDateOfDeath)"
-        }
-      ],
-      "objectName":"probateCaseDetails"
-    }
-  ],
-  "error":"Bad Request",
-  "status":400,
-  "message":"Validation failed for object='probateCaseDetails'. Error count: 1"
+  "probateCaseDetails": {
+    "caseData": {
+      "deceasedMartialStatus": "marriedCivilPartnership",
+      "primaryApplicantPhoneNumber": "123455678",
+      "childrenOverEighteenSurvived": "Yes",
+      "deceasedFreeTextAddress": "Winterfell, Westeros",
+      "deceasedAddressFound": "Yes",
+      "primaryApplicantAdoptionInEnglandOrWales": "Yes",
+      "uploadDocumentUrl": "http://document-management/document/12345",
+      "childrenDied": "No",
+      "deceasedSpouseNotApplyingReason": "mentallyIncapable",
+      "declaration": {
+        "declarationCheckbox": "Yes"
+      },
+      "primaryApplicantFreeTextAddress": "Pret a Manger St. Georges Hospital Blackshaw Road",
+      "ihtFormCompletedOnline": "Yes",
+      "deceasedAliasNameList": [
+        {
+          "value": {
+            "Forenames": "King",
+            "LastName": "North"
+          }
+        }
+      ],
+      "ihtGrossValue": "100000",
+      "applicationType": "Personal",
+      "ihtNetValue": "100000",
+      "deceasedAnyOtherNames": "No",
+      "ihtReferenceNumber": "GOT123456",
+      "deceasedAnyChildren": "No",
+      "outsideUKGrantCopies": 6,
+      "primaryApplicantRelationshipToDeceased": "adoptedChild",
+      "extraCopiesOfGrant": 5,
+      "deceasedForenames": "Ned",
+      "primaryApplicantEmailAddress": "jsnow@bbc.co.uk",
+      "primaryApplicantSurname": "Snow",
+      "primaryApplicantForenames": "Jon",
+      "primaryApplicantAddress": {
+        "AddressLine1": "Pret a Manger St. Georges Hospital Blackshaw Road London SW17 0QT"
+      },
+      "deceasedHasAssetsOutsideUK": "Yes",
+      "deceasedAddress": {
+        "AddressLine1": "Winterfell, Westeros"
+      },
+      "deceasedDivorcedInEnglandOrWales": "No",
+      "deceasedDateOfBirth": "2018-01-01",
+      "assetsOverseasNetValue": "10050",
+      "caseType": "intestacy",
+      "primaryApplicantAddressFound": "Yes",
+      "deceasedOtherChildren": "Yes",
+      "grandChildrenSurvivedUnderEighteen": "No",
+      "ihtFormId": "IHT205",
+      "type": "GrantOfRepresentation",
+      "registryLocation": "Birmingham",
+      "deceasedSurname": "Stark",
+      "deceasedDateOfDeath": "1920-01-01"
+    },
+    "caseInfo": {
+      "caseId": "12323213323",
+      "state": "Draft"
+    }
+  },
+  "validatorResults": {
+    "validationMessages": [
+      "DeceasedDateOfDeath before DeceasedDateOfBirth"
+    ],
+    "valid": false
+  }
 }
\ No newline at end of file
Index: build.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- build.gradle	(revision 1a78c098d76c9b6dcb1339ca978ed2f3de449dd8)
+++ build.gradle	(date 1548666943000)
@@ -319,7 +319,7 @@
 pact {
     publish {
         pactDirectory = 'target/pacts'
-        pactBrokerUrl = "$System.env.PACT_BROKER_URL"
+        pactBrokerUrl = "http://localhost:80"
         tags = [project.pactBrokerTag]
         version = [project.pactVersion]
     }
Index: src/main/java/uk/gov/hmcts/probate/core/service/mapper/IntestacyMapper.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/uk/gov/hmcts/probate/core/service/mapper/IntestacyMapper.java	(revision 1a78c098d76c9b6dcb1339ca978ed2f3de449dd8)
+++ src/main/java/uk/gov/hmcts/probate/core/service/mapper/IntestacyMapper.java	(date 1548412825000)
@@ -29,7 +29,7 @@
 
 
     @Mapping(target = "applicationType", expression = "java(ApplicationType.PERSONAL)")
-    @Mapping(target = "caseType", expression = "java(GrantType.INTESTACY)")
+    @Mapping(target = "grantType", expression = "java(GrantType.INTESTACY)")
     @Mapping(target = "primaryApplicantForenames", source = "applicant.firstName")
     @Mapping(target = "primaryApplicantSurname", source = "applicant.lastName")
     @Mapping(target = "primaryApplicantRelationshipToDeceased", source = "applicant.relationshipToDeceased")
Index: src/contractTest/resources/json/intestacyGrantOfRepresentation_payment.json
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/contractTest/resources/json/intestacyGrantOfRepresentation_payment.json	(revision 1a78c098d76c9b6dcb1339ca978ed2f3de449dd8)
+++ src/contractTest/resources/json/intestacyGrantOfRepresentation_payment.json	(date 1548672457000)
@@ -1,5 +1,6 @@
 {
   "caseType": "GrantOfRepresentation",
+  "caseType":"intestacy",
   "payment": {
     "status": "Success",
     "date": "2018-12-03",
Index: src/test/java/uk/gov/hmcts/probate/core/service/SubmitServiceImplTest.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/test/java/uk/gov/hmcts/probate/core/service/SubmitServiceImplTest.java	(revision 1a78c098d76c9b6dcb1339ca978ed2f3de449dd8)
+++ src/test/java/uk/gov/hmcts/probate/core/service/SubmitServiceImplTest.java	(date 1548763580000)
@@ -19,6 +19,7 @@
 import uk.gov.hmcts.reform.probate.model.cases.CollectionMember;
 import uk.gov.hmcts.reform.probate.model.cases.ProbateCaseDetails;
 import uk.gov.hmcts.reform.probate.model.cases.ProbatePaymentDetails;
+import uk.gov.hmcts.reform.probate.model.cases.SubmitResult;
 import uk.gov.hmcts.reform.probate.model.cases.grantofrepresentation.GrantOfRepresentationData;
 import uk.gov.hmcts.reform.probate.model.forms.Form;
 
@@ -122,7 +123,7 @@
     @Test
     public void shouldSubmit() {
         when(submitServiceApi.submit(eq(AUTHORIZATION), eq(SERVICE_AUTHORIZATION),
-            eq(EMAIL_ADDRESS), any(ProbateCaseDetails.class))).thenReturn(probateCaseDetails);
+            eq(EMAIL_ADDRESS), any(ProbateCaseDetails.class))).thenReturn(new SubmitResult(probateCaseDetails, null));
 
         Form formResponse = submitService.submit(EMAIL_ADDRESS, form);
 
Index: src/contractTest/resources/json/intestacyGrantOfRepresentation_invalid.json
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/contractTest/resources/json/intestacyGrantOfRepresentation_invalid.json	(revision 1a78c098d76c9b6dcb1339ca978ed2f3de449dd8)
+++ src/contractTest/resources/json/intestacyGrantOfRepresentation_invalid.json	(date 1548758237000)
@@ -1,6 +1,7 @@
 {
   "caseData": {
     "type": "GrantOfRepresentation",
+    "caseType":"intestacy",
     "applicationType": "Personal",
     "primaryApplicantEmailAddress": "jsnow@bbc.co.uk",
     "primaryApplicantForenames": "Jon",
@@ -55,7 +56,7 @@
     "uploadDocumentUrl": "http://document-management/document/12345"
   },
   "caseInfo": {
-    "caseId": "caseId",
-    "state": "state"
+    "caseId": "12323213323",
+    "state": "Draft"
   }
 }
Index: src/contractTest/resources/json/intestacyGrantOfRepresentation_invalid_presubmit.json
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/contractTest/resources/json/intestacyGrantOfRepresentation_invalid_presubmit.json	(date 1548772721000)
+++ src/contractTest/resources/json/intestacyGrantOfRepresentation_invalid_presubmit.json	(date 1548772721000)
@@ -0,0 +1,62 @@
+{
+  "caseData": {
+    "type": "GrantOfRepresentation",
+    "caseType":"intestacy",
+    "applicationType": "Personal",
+    "primaryApplicantEmailAddress": "jsnow@bbc.co.uk",
+    "primaryApplicantForenames": "Jon",
+    "primaryApplicantSurname": "Snow",
+    "primaryApplicantAddress": {
+      "AddressLine1": "Pret a Manger St. Georges Hospital Blackshaw Road London SW17 0QT"
+    },
+    "primaryApplicantFreeTextAddress": "Pret a Manger St. Georges Hospital Blackshaw Road",
+    "primaryApplicantAddressFound": "Yes",
+    "primaryApplicantPhoneNumber": "123455678",
+    "primaryApplicantRelationshipToDeceased": "adoptedChild",
+    "primaryApplicantAdoptionInEnglandOrWales": "Yes",
+    "deceasedForenames": "N",
+    "deceasedSurname": "Stark",
+    "deceasedDateOfBirth": "1910-01-01",
+    "deceasedDateOfDeath": "2018-01-01",
+    "deceasedAddress": {
+      "AddressLine1": "Winterfell, Westeros"
+    },
+    "deceasedFreeTextAddress": "Winterfell, Westeros",
+    "deceasedAddressFound": "Yes",
+    "deceasedAnyOtherNames": "No",
+    "deceasedAliasNameList": [
+      {
+        "value": {
+          "Forenames": "King",
+          "LastName": "North"
+        }
+      }
+    ],
+    "deceasedMartialStatus": "marriedCivilPartnership",
+    "deceasedDivorcedInEnglandOrWales": "No",
+    "deceasedSpouseNotApplyingReason": "mentallyIncapable",
+    "deceasedOtherChildren": "Yes",
+    "childrenOverEighteenSurvived": "Yes",
+    "childrenDied": "No",
+    "grandChildrenSurvivedUnderEighteen": "No",
+    "deceasedAnyChildren": "No",
+    "registryLocation": "Birmingham",
+    "assetsOverseasNetValue": "10050",
+    "deceasedHasAssetsOutsideUK": "Yes",
+    "ihtFormId": "IHT205",
+    "ihtFormCompletedOnline": "Yes",
+    "ihtGrossValue": "100000",
+    "ihtNetValue": "100000",
+    "ihtReferenceNumber": "GOT123456",
+    "declaration": {
+      "declarationCheckbox": "Yes"
+    },
+    "extraCopiesOfGrant": 5,
+    "outsideUKGrantCopies": 6,
+    "uploadDocumentUrl": "http://document-management/document/12345"
+  },
+  "caseInfo": {
+    "caseId": "12323213323",
+    "state": "Draft"
+  }
+}
Index: src/contractTest/resources/json/intestacyGrantOfRepresentation_invalid_presubmit_response.json
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/contractTest/resources/json/intestacyGrantOfRepresentation_invalid_presubmit_response.json	(date 1548775156000)
+++ src/contractTest/resources/json/intestacyGrantOfRepresentation_invalid_presubmit_response.json	(date 1548775156000)
@@ -0,0 +1,1 @@
+
Index: src/contractTest/resources/json/intestacyGrantOfRepresentation_full.json
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/contractTest/resources/json/intestacyGrantOfRepresentation_full.json	(revision 1a78c098d76c9b6dcb1339ca978ed2f3de449dd8)
+++ src/contractTest/resources/json/intestacyGrantOfRepresentation_full.json	(date 1548757598000)
@@ -1,6 +1,7 @@
 {
   "caseData": {
     "type": "GrantOfRepresentation",
+    "caseType": "intestacy",
     "applicationType": "Personal",
     "primaryApplicantEmailAddress": "jsnow@bbc.co.uk",
     "primaryApplicantForenames": "Jon",
@@ -55,7 +56,7 @@
     "uploadDocumentUrl": "http://document-management/document/12345"
   },
   "caseInfo": {
-    "caseId": "caseId",
-    "state": "state"
+    "caseId": "12323213323",
+    "state": "Draft"
   }
 }
Index: src/contractTest/java/uk/gov/hmcts/probate/contract/tests/SubmitServiceConsumerCaseDetailsTest.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/contractTest/java/uk/gov/hmcts/probate/contract/tests/SubmitServiceConsumerCaseDetailsTest.java	(revision 1a78c098d76c9b6dcb1339ca978ed2f3de449dd8)
+++ src/contractTest/java/uk/gov/hmcts/probate/contract/tests/SubmitServiceConsumerCaseDetailsTest.java	(date 1548776166000)
@@ -18,6 +18,7 @@
 import uk.gov.hmcts.probate.client.SubmitServiceApi;
 import uk.gov.hmcts.reform.probate.model.cases.CaseType;
 import uk.gov.hmcts.reform.probate.model.cases.ProbateCaseDetails;
+import uk.gov.hmcts.reform.probate.model.cases.SubmitResult;
 
 import java.io.File;
 import java.io.FileNotFoundException;
@@ -44,7 +45,8 @@
 
     public static final String SOME_AUTHORIZATION_TOKEN = "someAuthorizationToken";
     public static final String SOME_SERVICE_AUTHORIZATION_TOKEN = "someServiceAuthorizationToken";
-    public static final String SOMEEMAILADDRESS_HOST_COM = "email";
+    public static final String SOMEEMAILADDRESS_HOST_COM = "jsnow@bbc.co.uk";
+    public static final String CASE_ID = "12323213323";
     @Autowired
     private SubmitServiceApi submitServiceApi;
     @Autowired
@@ -144,13 +146,14 @@
                 .willRespondWith()
                 .status(200)
                 .matchHeader("Content-Type", "application/json;charset=UTF-8")
-                .body(createJsonObject("intestacyGrantOfRepresentation_full.json"))
+                .body(createJsonObject("intestacyGrantOfRepresentation_full_response.json"))
                 .toPact();
         // @formatter:on
     }
 
     @Pact(state = "provider POSTS submission with validation errors", provider = "probate_submitservice_submissions", consumer = "probate_orchestratorservice_submitserviceclient")
-    public RequestResponsePact executePostSubmissionWithValidationErrorsDsl(PactDslWithProvider builder) throws IOException, JSONException {
+    public RequestResponsePact executePostSubmissionWithValidationErrors(PactDslWithProvider builder) throws IOException, JSONException {
+        // @formatter:off
         return builder
                 .given("provider POSTS submission with validation errors")
                 .uponReceiving("a request to POST an invalid submission")
@@ -162,102 +165,60 @@
                 .willRespondWith()
                 .status(400)
                 .matchHeader("Content-Type", "application/json;charset=UTF-8")
+                .body(createJsonObject("intestacyGrantOfRepresentation_invalid_response.json"))
+                .toPact();
+        // @formatter:on
+    }
+
+
+    @Pact(state = "provider POSTS submission with presubmit validation errors", provider = "probate_submitservice_submissions", consumer = "probate_orchestratorservice_submitserviceclient")
+    public RequestResponsePact executePostSubmissionWithValidationErrorsDsl(PactDslWithProvider builder) throws IOException, JSONException {
+        return builder
+                .given("provider POSTS submission with presubmit validation errors")
+                .uponReceiving("a request to POST an invalid submission")
+                .path("/submissions/" + SOMEEMAILADDRESS_HOST_COM)
+                .method("POST")
+                .headers(AUTHORIZATION, SOME_AUTHORIZATION_TOKEN, SERVICE_AUTHORIZATION, SOME_SERVICE_AUTHORIZATION_TOKEN)
+                .matchHeader("Content-Type", "application/json")
+                .body(createJsonObject("intestacyGrantOfRepresentation_invalid_presubmit.json"))
+                .willRespondWith()
+                .status(400)
+                .matchHeader("Content-Type", "application/json;charset=UTF-8")
                 .body(newJsonBody((o) -> {
-                    o.stringValue("path", "/submissions/email");
+                    o.stringValue("path", "/submissions/" + SOMEEMAILADDRESS_HOST_COM);
                     o.stringType("timestamp");
                     o.array("errors", (a) -> {
-                                a.object((e) -> e.stringValue("field", "caseData")
+                                a.object((e) -> e.stringValue("field", "caseData.deceasedForenames")
                                         .booleanValue("bindingFailure", false)
-                                        .stringValue("code", "AssertExpression")
+                                        .stringValue("code", "Size")
                                         .array("codes", (c) -> {
-                                            c.stringValue("AssertExpression.probateCaseDetails.caseData")
-                                                    .stringValue("AssertExpression.caseData")
-                                                    .stringValue("AssertExpression.uk.gov.hmcts.reform.probate.model.cases.CaseData")
-                                                    .stringValue("AssertExpression");
+                                            c.stringValue("Size.probateCaseDetails.caseData.deceasedForenames")
+                                                    .stringValue("Size.caseData.deceasedForenames")
+                                                    .stringValue("Size.deceasedForenames")
+                                                    .stringValue("Size.java.lang.String")
+                                                    .stringValue("Size");
                                         })
-                                        .stringValue("defaultMessage", "deceasedDateOfBirth.isBefore(deceasedDateOfDeath) is false")
-                                        .object("rejectedValue", (r) -> r.stringValue("primaryApplicantPhoneNumber", "123455678")
-                                                .stringValue("deceasedFreeTextAddress", "Winterfell, Westeros")
-                                                .stringValue("deceasedAddressFound", "Yes")
-                                                .stringValue("primaryApplicantAdoptionInEnglandOrWales", "Yes")
-                                                .stringValue("uploadDocumentUrl", "http://document-management/document/12345")
-                                                .stringValue("deceasedSpouseNotApplyingReason", "mentallyIncapable")
-                                                .object("declaration", (d) -> d.stringValue("declarationCheckbox", "Yes"))
-                                                .stringValue("primaryApplicantFreeTextAddress", "Pret a Manger St. Georges Hospital Blackshaw Road")
-                                                .stringValue("childrenDied", "No")
-                                                .stringValue("ihtFormCompletedOnline", "Yes")
-                                                .array("deceasedAliasNameList", (n) -> {
-                                                    n.object((f) -> f.object("value", (g) -> {
-                                                        g.stringValue("Forenames", "King")
-                                                                .stringValue("LastName", "North");
-                                                    }));
-
-                                                })
-                                                .stringValue("ihtGrossValue", "100000")
-                                                .stringValue("applicationType", "Personal")
-                                                .stringValue("ihtNetValue", "100000")
-                                                .stringValue("deceasedAnyOtherNames", "No")
-                                                .stringValue("ihtReferenceNumber", "GOT123456")
-                                                .stringValue("deceasedHasAssetsOutsideUK", "Yes")
-                                                .stringValue("deceasedAnyChildren", "No")
-                                                .numberValue("outsideUKGrantCopies", 6)
-                                                .stringValue("primaryApplicantRelationshipToDeceased", "adoptedChild")
-                                                .numberValue("extraCopiesOfGrant", 5)
-                                                .stringValue("deceasedForenames", "Ned")
-                                                .stringValue("primaryApplicantEmailAddress", "jsnow@bbc.co.uk")
-                                                .stringValue("primaryApplicantSurname", "Snow")
-                                                .stringValue("primaryApplicantForenames", "Jon")
-                                                .object("primaryApplicantAddress", (p) -> p.stringValue("AddressLine1", "Pret a Manger St. Georges Hospital Blackshaw Road London SW17 0QT"))
-                                                .object("deceasedAddress", (p) -> p.stringValue("AddressLine1", "Winterfell, Westeros"))
-                                                .stringValue("deceasedDivorcedInEnglandOrWales", "No")
-                                                .stringValue("deceasedMartialStatus", "marriedCivilPartnership")
-                                                .stringValue("deceasedDateOfBirth", "2018-01-01")
-                                                .stringValue("assetsOverseasNetValue", "10050")
-                                                .stringValue("primaryApplicantAddressFound", "Yes")
-                                                .stringValue("deceasedOtherChildren", "Yes")
-                                                .stringValue("ihtFormId", "IHT205")
-                                                .stringValue("registryLocation", "Birmingham")
-                                                .stringValue("childrenOverEighteenSurvived", "Yes")
-                                                .stringValue("deceasedSurname", "Stark")
-                                                .stringValue("grandChildrenSurvivedUnderEighteen", "No")
-                                                .stringValue("deceasedDateOfDeath", "1920-01-01")
+                                        .stringValue("defaultMessage", "size must be between 2 and 2147483647")
+                                        .stringValue("rejectedValue", ("N")
                                         ));
                             }
 
                     );
                     o.stringValue("error", "Bad Request");
                     o.numberValue("status", 400);
-                    o.stringValue("message","Validation failed for object='probateCaseDetails'. Error count: 1");
+                    o.stringValue("message", "Validation failed for object='probateCaseDetails'. Error count: 1");
                 }).build())
                 .toPact();
 
     }
 
-//    @Pact(state = "provider POSTS submission with validation errors", provider = "probate_submitservice_submissions", consumer = "probate_orchestratorservice_submitserviceclient")
-//    public RequestResponsePact executePostSubmissionWithValidationErrors(PactDslWithProvider builder) throws IOException, JSONException {
-//        // @formatter:off
-//        return builder
-//                .given("provider POSTS submission with validation errors")
-//                .uponReceiving("a request to POST an invalid submission")
-//                .path("/submissions/" + SOMEEMAILADDRESS_HOST_COM)
-//                .method("POST")
-//                .headers(AUTHORIZATION, SOME_AUTHORIZATION_TOKEN, SERVICE_AUTHORIZATION, SOME_SERVICE_AUTHORIZATION_TOKEN)
-//                .matchHeader("Content-Type", "application/json")
-//                .body(createJsonObject("intestacyGrantOfRepresentation_invalid.json"))
-//                .willRespondWith()
-//                .status(400)
-//                .matchHeader("Content-Type", "application/json;charset=UTF-8")
-//                .body(createJsonObject("intestacyGrantOfRepresentation_invalid_response.json"))
-//                .toPact();
-//        // @formatter:on
-//    }
 
     @Test
     @PactTestFor(pactMethod = "executeSuccessGetCaseDataPact")
     public void verifyExecuteSuccessGetCaseDataPact() {
 
         ProbateCaseDetails caseDetails = submitServiceApi.getCase(SOME_AUTHORIZATION_TOKEN, SOME_SERVICE_AUTHORIZATION_TOKEN, SOMEEMAILADDRESS_HOST_COM, CaseType.GRANT_OF_REPRESENTATION.toString());
-        assertThat(caseDetails.getCaseInfo().getCaseId(), equalTo("caseId"));
+        assertThat(caseDetails.getCaseInfo().getCaseId(), equalTo(CASE_ID));
     }
 
     @Test
@@ -265,7 +226,7 @@
     public void verifyExecuteSuccessPostPartialDraftCaseDataPact() throws IOException, JSONException {
 
         ProbateCaseDetails caseDetails = submitServiceApi.saveDraft(SOME_AUTHORIZATION_TOKEN, SOME_SERVICE_AUTHORIZATION_TOKEN, SOMEEMAILADDRESS_HOST_COM, getProbateCaseDetails("intestacyGrantOfRepresentation_partial_draft.json"));
-        assertThat(caseDetails.getCaseInfo().getCaseId(), equalTo("caseId"));
+        assertThat(caseDetails.getCaseInfo().getCaseId(), equalTo(CASE_ID));
     }
 
     @Test
@@ -273,15 +234,7 @@
     public void verifyExecuteSuccessPostDraftCaseDataPact() throws IOException, JSONException {
 
         ProbateCaseDetails caseDetails = submitServiceApi.saveDraft(SOME_AUTHORIZATION_TOKEN, SOME_SERVICE_AUTHORIZATION_TOKEN, SOMEEMAILADDRESS_HOST_COM, getProbateCaseDetails("intestacyGrantOfRepresentation_full.json"));
-        assertThat(caseDetails.getCaseInfo().getCaseId(), equalTo("caseId"));
-    }
-
-    @Test
-    @PactTestFor(pactMethod = "executePostSubmissionWithSuccessPact")
-    public void verifyExecutePostSubmissionWithSuccessPact() throws IOException, JSONException {
-
-        ProbateCaseDetails caseDetails = submitServiceApi.submit(SOME_AUTHORIZATION_TOKEN, SOME_SERVICE_AUTHORIZATION_TOKEN, SOMEEMAILADDRESS_HOST_COM, getProbateCaseDetails("intestacyGrantOfRepresentation_full.json"));
-        assertThat(caseDetails.getCaseInfo().getCaseId(), equalTo("caseId"));
+        assertThat(caseDetails.getCaseInfo().getCaseId(), equalTo(CASE_ID));
     }
 
     @Test
@@ -293,9 +246,16 @@
 
     }
 
-
     @Test
-    @PactTestFor(pactMethod = "executePostSubmissionWithValidationErrorsDsl")
+    @PactTestFor(pactMethod = "executePostSubmissionWithSuccessPact")
+    public void verifyExecutePostSubmissionWithSuccessPact() throws IOException, JSONException {
+
+        SubmitResult submitResult = submitServiceApi.submit(SOME_AUTHORIZATION_TOKEN, SOME_SERVICE_AUTHORIZATION_TOKEN, SOMEEMAILADDRESS_HOST_COM, getProbateCaseDetails("intestacyGrantOfRepresentation_full.json"));
+        assertThat(submitResult.getProbateCaseDetails().getCaseInfo().getCaseId(), equalTo(CASE_ID));
+    }
+
+    @Test
+    @PactTestFor(pactMethod = "executePostSubmissionWithValidationErrors")
     public void verifyExecutePostSubmissionWithValidationErrors() throws IOException, JSONException {
         assertThrows(FeignException.class, () -> {
             submitServiceApi.submit(SOME_AUTHORIZATION_TOKEN, SOME_SERVICE_AUTHORIZATION_TOKEN, SOMEEMAILADDRESS_HOST_COM, getProbateCaseDetails("intestacyGrantOfRepresentation_invalid.json"));
@@ -303,6 +263,15 @@
 
     }
 
+    @Test
+    @PactTestFor(pactMethod = "executePostSubmissionWithValidationErrorsDsl")
+    public void verifyExecutePostSubmissionWithValidationErrorsDsl() throws IOException, JSONException {
+        assertThrows(FeignException.class, () -> {
+            submitServiceApi.submit(SOME_AUTHORIZATION_TOKEN, SOME_SERVICE_AUTHORIZATION_TOKEN, SOMEEMAILADDRESS_HOST_COM, getProbateCaseDetails("intestacyGrantOfRepresentation_invalid_presubmit.json"));
+        });
+
+    }
+
     private JSONObject createJsonObject(String fileName) throws JSONException, IOException {
         File file = getFile(fileName);
         String jsonString = new String(Files.readAllBytes(file.toPath()));
Index: src/contractTest/resources/json/intestacyGrantOfRepresentation_full_response.json
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/contractTest/resources/json/intestacyGrantOfRepresentation_full_response.json	(date 1548768655000)
+++ src/contractTest/resources/json/intestacyGrantOfRepresentation_full_response.json	(date 1548768655000)
@@ -0,0 +1,69 @@
+{
+  "probateCaseDetails": {
+    "caseData": {
+      "deceasedMartialStatus": "marriedCivilPartnership",
+      "primaryApplicantPhoneNumber": "123455678",
+      "childrenOverEighteenSurvived": "Yes",
+      "deceasedFreeTextAddress": "Winterfell, Westeros",
+      "deceasedAddressFound": "Yes",
+      "primaryApplicantAdoptionInEnglandOrWales": "Yes",
+      "uploadDocumentUrl": "http://document-management/document/12345",
+      "childrenDied": "No",
+      "deceasedSpouseNotApplyingReason": "mentallyIncapable",
+      "declaration": {
+        "declarationCheckbox": "Yes"
+      },
+      "primaryApplicantFreeTextAddress": "Pret a Manger St. Georges Hospital Blackshaw Road",
+      "ihtFormCompletedOnline": "Yes",
+      "deceasedAliasNameList": [
+        {
+          "value": {
+            "Forenames": "King",
+            "LastName": "North"
+          }
+        }
+      ],
+      "ihtGrossValue": "100000",
+      "applicationType": "Personal",
+      "ihtNetValue": "100000",
+      "deceasedAnyOtherNames": "No",
+      "ihtReferenceNumber": "GOT123456",
+      "deceasedAnyChildren": "No",
+      "outsideUKGrantCopies": 6,
+      "primaryApplicantRelationshipToDeceased": "adoptedChild",
+      "extraCopiesOfGrant": 5,
+      "deceasedForenames": "Ned",
+      "primaryApplicantEmailAddress": "jsnow@bbc.co.uk",
+      "primaryApplicantSurname": "Snow",
+      "primaryApplicantForenames": "Jon",
+      "primaryApplicantAddress": {
+        "AddressLine1": "Pret a Manger St. Georges Hospital Blackshaw Road London SW17 0QT"
+      },
+      "deceasedHasAssetsOutsideUK": "Yes",
+      "deceasedAddress": {
+        "AddressLine1": "Winterfell, Westeros"
+      },
+      "deceasedDivorcedInEnglandOrWales": "No",
+      "deceasedDateOfBirth": "1930-01-01",
+      "assetsOverseasNetValue": "10050",
+      "caseType": "intestacy",
+      "primaryApplicantAddressFound": "Yes",
+      "deceasedOtherChildren": "Yes",
+      "grandChildrenSurvivedUnderEighteen": "No",
+      "ihtFormId": "IHT205",
+      "type": "GrantOfRepresentation",
+      "registryLocation": "Birmingham",
+      "deceasedSurname": "Stark",
+      "deceasedDateOfDeath": "2018-01-01"
+    },
+    "caseInfo": {
+      "caseId": "12323213323",
+      "state": "Draft"
+    }
+  },
+  "validatorResults": {
+    "validationMessages": [
+    ],
+    "valid": true
+  }
+}
\ No newline at end of file
